test/conftest.py

```diff
@@ -147,33 +174,49 @@ def test_workspaces(
     )
 
 
-@pytest.fixture(scope="session", autouse=True)
-def ws_id(test_workspaces: dict[str, Any]) -> int:
-    return test_workspaces["ws_id"]
+@pytest.fixture(scope="session")
+def default_ws_id(test_workspaces: dict[str, Any]) -> int:
+    """ID for the default test workspace."""
+    return test_workspaces["default_ws_id"]
 
 
-@pytest.fixture(scope="session", autouse=True)
-def ws_name(test_workspaces: dict[str, Any]) -> str:
-    return test_workspaces["ws_name"]
+@pytest.fixture(scope="session")
+def default_ws_name(test_workspaces: dict[str, Any]) -> str:
+    """Name of the default test workspace."""
+    return test_workspaces["default_ws_name"]
 
 
 @pytest.fixture(scope="session")
 def list_all_sets_ws_name(test_workspaces: dict[str, Any]) -> int:
+    """Name of the 'list_all_sets' test workspace."""
     return test_workspaces["list_all_sets_ws_name"]
 
 
 @pytest.fixture(scope="session")
 def list_all_sets_ws_id(test_workspaces: dict[str, Any]) -> int:
+    """ID for the 'list_all_sets' test workspace."""
     return test_workspaces["list_all_sets_ws_id"]
 
 
-@pytest.fixture(scope="session", autouse=True)
+@pytest.fixture(scope="session", params=["default_ws_id", "list_all_sets_ws_id"])
+def ws_id(request: FixtureRequest) -> int:
+    """Magical pytest function to choose the appropriate ws_id value."""
+    return request.getfixturevalue(request.param)
+
+
+@pytest.fixture(scope="session", params=["default_ws_name", "list_all_sets_ws_name"])
+def ws_name(request: FixtureRequest) -> str:
+    """Magical pytest function to choose the corresponding ws name."""
+    return request.getfixturevalue(request.param)
```

Comment:
these generic names are now fixtures with possible values for the default and list_all_sets versions.